core/src/main/kotlin/com/katanox/tabour/sqs/SqsRegistry.kt

```diff
@@ -0,0 +1,79 @@
+package com.katanox.tabour.sqs
+
+import com.katanox.tabour.configuration.Registry
+import com.katanox.tabour.sqs.config.SqsConsumer
+import com.katanox.tabour.sqs.consumption.SqsPoller
+import com.katanox.tabour.sqs.production.SqsProducer
+import com.katanox.tabour.sqs.production.SqsProducerExecutor
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+
+/**
+ * A type of [Registry] which works with SQS
+ *
+ * [T] is the key of the registry which is used to identify the registry
+ *
+ * Supports consumption and production of SQS messages
+ */
+class SqsRegistry<T>
+internal constructor(
+    private val configuration: Configuration<T>,
+) : Registry<T> {
+    override val key: T
+        get() = configuration.key
+
+    private val consumers: MutableList<SqsConsumer> = mutableListOf()
+    private val producers: MutableSet<SqsProducer<*>> = mutableSetOf()
+    private val sqs: SqsAsyncClient =
```

Comment:
Suggestion: In the case of creating multiple sqs registries that would result in creating multiple sqs clients which is not a good approach to create multiples of the client.

Reply:
we should create multiple clients, because those clients will need different credentials